services/meta/client.go

```diff
@@ -343,24 +343,24 @@ func (c *Client) CreateDatabaseWithRetentionPolicy(name string, rpi *RetentionPo
 		return nil, ErrRetentionPolicyDurationTooLow
 	}
 
-	if _, err := c.CreateDatabase(name); err != nil {
-		return nil, err
-	}
-
-	if err := c.DropRetentionPolicy(name, rpi.Name); err != nil {
-		return nil, err
+	if db, _ := c.Database(name); db != nil {
+		// Check if the retention policy already exists. If it does and matches
+		// the desired retention policy, exit with no error.
+		if rp := db.RetentionPolicy(rpi.Name); rp != nil {
+			if rp.ReplicaN != rpi.ReplicaN || rp.Duration != rpi.Duration {
+				return nil, ErrRetentionPolicyExists
```

Comment:
Could we return a slightly different error here?  When I hit this via the cli I get:
```
> create database if not exists test with duration 10h replication 1
> create database if not exists test with duration 15h replication 1
ERR: retention policy already exists
```

which is a little confusing because the retention policy does already exist, but it's different than what I'm trying to create.  Maybe the error text just needs to be updated.